tsdapiclient/tacl.py

```diff
@@ -619,7 +642,21 @@ def cli(
                 debug_step("API key has expired")
                 api_key = renew_api_key(env, pnum, api_key, key_file)
         debug_step('using basic authentication')
-        token, refresh_token = get_jwt_basic_auth(env, pnum, api_key, token_type)
+        if link_id:
+
+            if link_id.startswith("https://"):
+                click.echo("extracting link_id from URL")
+                patten = r"https://(?P<HOST>.+)/(?P<instance_type>c|i)/(?P<link_id>[a-f\d0-9-]{36})"
+                if match:=re.compile(patten).match(link_id):
+                    link_id = uuid.UUID(match.group("link_id"))
+                    if match.group("instance_type") == "c" and not secret_challenge:
+                        click.echo("instance authentication requires a secret challenge")
```

Comment:
We could consider interactively asking for the secret here

Reply:
Yes did not have time on Friday